src/test/scala/tech/cryptonomic/nautilus/cloud/adapters/inmemory/InMemoryApiKeyRepository.scala

```diff
@@ -115,4 +115,13 @@ class InMemoryApiKeyRepository[F[_]: Monad] extends ApiKeyRepository[F] {
       .pure[F]
   }
 
+  /** Gets keys which were active in between dates */
+  override def getUserActiveKeysInGivenRange(userId: UserId, start: Instant, end: Instant): F[List[ApiKey]] =
```

Comment:
Just thinking, isn't this getting cumbersome and non-trivial to keep up?

How does this implementation interact with the sub-systems being tested?

How easy is it to make a mistake here and not noticing until some unexpected bug shows up in deployment?

Sometimes it might be more direct to allow injecting specific custom behaviour from the test itself when building the in-memory-test-double (like mocks do). That might make the expectations for a single test easy to make out, independently of the data available in-memory and how it's actually collected.

Look at the comment in the `MeteringApiProcessorTest`

Reply:
I agree that we should possibly look at the other ways of testing, possibly look into writing test cases involving real DB(not whole integration tests cases, less complex).